### PR TITLE
refactor: use `CREATE` instead of `CREATE2`

### LINKED ISSUE #337: Use CREATE instead of CREATE2 when deploying the MerkleLockup contract
After the latest audit, some issues were found if the contract is pre-funded before deployment, e.g.: https://github.com/cantinasec/review-sablier-3/issues/1. Therefore, having deterministic contracts is problematic, and we should switch to using `CREATE`.

It will require to remove this:

https://github.com/sablier-labs/v2-periphery/blob/29be29e08ddd8c86e35252d76c1f81c7bf3db21e/src/SablierV2MerkleStreamerFactory.sol#L36-L51

### PR DESCRIPTION
**Pull request by smol-ninja**
Fixes #337.

### CODE REVIEW

File: test/fork/merkle-lockup/MerkleLL.t.sol [hunk 1]
```diff
@@ -95,7 +95,7 @@ abstract contract MerkleLL_Fork_Test is Fork_Test {
         MerkleBuilder.sortLeaves(leaves);
         vars.merkleRoot = getRoot(leaves.toBytes32());
 
-        vars.expectedLL = computeMerkleLLAddress(params.admin, FORK_ASSET, vars.merkleRoot, params.expiration);
+        vars.expectedLL = computeCreateAddress(address(merkleLockupFactory), ++merkleLockupFactoryNonce);
```
Comment (andreivladbrg) [inline test/fork/merkle-lockup/MerkleLL.t.sol:98]: let's use `vm. computeCreateAddress`

https://github.com/foundry-rs/forge-std/blob/bb4ceea94d6f10eeb5b41dc2391c6c8bf8e734ef/src/StdUtils.sol#L102

File: test/fork/merkle-lockup/MerkleLT.t.sol [hunk 2]
```diff
@@ -96,7 +96,7 @@ abstract contract MerkleLT_Fork_Test is Fork_Test {
         MerkleBuilder.sortLeaves(leaves);
         vars.merkleRoot = getRoot(leaves.toBytes32());
 
-        vars.expectedLT = computeMerkleLTAddress(params.admin, FORK_ASSET, vars.merkleRoot, params.expiration);
+        vars.expectedLT = computeCreateAddress(address(merkleLockupFactory), ++merkleLockupFactoryNonce);
```
Comment (andreivladbrg) [inline test/fork/merkle-lockup/MerkleLT.t.sol:99]: same as above

**Review by andreivladbrg — APPROVED** [review]
LGTM 👌

**Comment by andreivladbrg** [comment]
i've tried to run the fork tests locally, it seems to not work for me, is it the same for you? @smol-ninja

**Comment by smol-ninja** [comment]
> i've tried to run the fork tests locally, it seems to not work for me

Yes its RPC endpoint issue. If you run them individually, it works.

**Pull request merged**